chrome-extension/popup.js

```diff
@@ -129,28 +144,88 @@ function generateNote() {
       name: docName,
     }
   }).then(response => {
-    // TODO: Make POST request to upload notes servlet
-    const gNoteURL = GOOGLE_DOC_URL + response.result.id;
-    loadingIcon.style.display = 'none';
-    chrome.tabs.create({ url: gNoteURL });
+    const docId = response.result.id;
+    const gNoteUrl = GOOGLE_DOC_URL + docId;
+    compileNoteData(docName, docId, gNoteUrl, school, course);
   }).catch(error => {
-    loadingIcon.style.display = 'none';
+    setLoadingIcon(false);
     console.log('Error:', error);
   })
 }
 
+/** Retrieves input values and sets them if they are blank */
+function retrieveParams() {
+  let docName = document.getElementById('doc-name-input').value.trim();
+  if(docName === '') docName = 'gNote ' + getDate();
+  let school = document.getElementById('school-input').value.trim();
+  if(school === '') school = 'Unaffiliated';
+  let course = document.getElementById('course-input').value.trim();
+  if(course === '') course = 'Unaffiliated';
+  return [docName, school, course];
+}
+
+/** Puts together the URL search params for posting the note */
+async function compileNoteData(title, docId, docUrl, school, course) {
+  getPDFLink(docId)
+    .then(response => {
+      const pdfSource = response.result.exportLinks['application/pdf'];
+      const payload = {
+        title: title,
+        school: school,
+        course: course,
+        sourceUrl: docUrl,
+        pdfSource: pdfSource
+      }
+      const noteData = new URLSearchParams(payload);
+      postNoteToDatabase(noteData)
```

Comment:
nit: missing semicolon

Reply:
oops my bad I keep forgetting semicolons

Reply:
Wait does there need to be a semicolon here if there is a `.then()`?

Reply:
Ohhh no we don't, I didn't realize this `.then()` was a nested `.then()` haha